code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -196,6 +199,10 @@
 			use_power = on ? ACTIVE_POWER_USE : IDLE_POWER_USE
 			investigate_log("was turned [on ? "on" : "off"] by [key_name(usr)]", INVESTIGATE_ATMOS)
 			. = TRUE
+			if(on == TRUE)
+				was_on = TRUE   //if the machine was manually turned on, ensure it remembers it
+			else
+				was_on = FALSE  //else dont remember it was on, because it wasnt.
```

Comment:
```suggestion
			was_on = on
```

Reply:
that wont work though, because then hitting the button to turn it off will also set it to was_on, when it actually needs to set it back to false when turned off

Reply:
This is exactly the same as your code, unless on is for whatever reason not always TRUE/FALSE.

Please recheck your code.

Reply:
changed to `was_on = !was_on`